Oracle.md

```diff
@@ -0,0 +1,165 @@
+### Serialization of event descriptors
+
+Event descriptors should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) as described bellow.
+
+#### Version 0 `external_event_descriptor`
+
+1. type: 55300 (`external_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`string`:`external_name`]
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`u16`:`num_outcomes`]
+   * [`u16`:`outcome_1_len`]
+   * [`string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
+
+Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+
+#### Version 0 `range_event_descriptor`
+
+1. type: 55304 (`range_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`int32`:`start`]
+   * [`u32`:`count`]
+   * [`u16`:`step`]
+
+#### Version 0 `large_range_event_descriptor`
+
+1. type: 55306 (`large_range_event_descriptor_v0`)
+2. data:
+   * [`u16`:`base`]
+   * [`byte`:`is_signed`]
+   * [`u16`:`nb_nonces`]
+   * [`nb_nonces*x_point`:`oracle_nonces`]
+
+`is_signed` should be set to `0x00` for representing false and `0x01` for representing true.
+Any other value for it should be considered invalid.
+
+### Oracle events
+
+For users to be able to create DLCs based on a given event, they also need to obtain information about the oracle and the time at which it plans on releasing a signature over the event outcome.
+Oracle events contain such information, which includes:
+* the oracle public key,
+* the earliest time (UTC) at which it plans on releasing a signature over the event outcome, in epoch seconds,
+* the event descriptor,
+* the event URI which can be a name or categorization associated with the event by the oracle.
+
+The TLV serialization for oracle events is as follow:
+
+#### Version 0 `oracle_event`
+
+1. type: 55330 (`oracle_event_v0`)
+2. data:
+   * [`x_point`:`oracle_public_key`]
+   * [`u32`:`event_maturity_epoch`]
+   * [`event_descriptor`:`event_descriptor`]
+   * [`string`:`event_uri`]
+
+## Oracle announcements
+
+In order to make it possible to hold oracles accountable in cases where they do not release a signature for an event outcome, there needs to be a proof that an oracle has committed to a given outcome.
+This proof is given in a so-called oracle announcement, which contains an oracle event together with a signature over its serialization, created using the same private key that the oracle plans on signing the event outcome with.
+
+This also makes it possible for users to obtain oracle event information from an un-trusted peer while being guaranteed that it originates from a given oracle.
+
+The TLV serialization of oracle announcements is as follow.
```

Comment:
Unlike all the other things in this doc, the Oracle Announcement should be a message and serialized as an LN message, not a TLV

Reply:
I'm on the fence about this one now. If we want to send this (by itself) over the wire then it should be a message, but if this is used (P2P) primarily only within offers (as I propose in #108), then it should be a TLV

Reply:
Yeah I agree it can be both, maybe we should keep it as a TLV here and define a message that just wraps it?

Reply:
I am planning to make a PR that embeds the oracle announcement in a contract info (which is itself embedded in an offer message) so I'm cool with leaving it a TLV